api/query/search.go

```diff
@@ -109,8 +121,26 @@ func (sh structuredSearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 
 	simpleQ, ok := rq.AbstractQuery.(TestNamePattern)
 	if !ok {
-		// TODO: Implement serving complex queries.
-		http.Error(w, "Query pattern is valid, but processing of this query configuration is not yet implemented", http.StatusServiceUnavailable)
+		ctx := r.Context()
+		hostname := sh.api.GetHostname()
+		url := fmt.Sprintf("https://%s/api/search/cache", hostname)
+
+		shared.GetLogger(ctx).Infof("Forwarding structured search request to cache: %s", string(data))
+
+		client := sh.api.GetHTTPClient()
+		resp, err := client.Post(url, "application/json", bytes.NewBuffer(data))
+		if err != nil {
+			http.Error(w, "Error connecting to search API cache", http.StatusInternalServerError)
+			return
+		}
+
+		data, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			http.Error(w, "Error reading response from search API cache", http.StatusInternalServerError)
+		}
+
+		w.WriteHeader(resp.StatusCode)
+		w.Write(data)
```

Comment:
nit: You can collapse a read all + write into a `Copy`, something like

    io.Copy(w, resp.Body)